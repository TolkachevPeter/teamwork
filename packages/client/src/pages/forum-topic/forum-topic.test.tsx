import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import ForumTopicPage from './forum-topic-page'

const mockUsedNavigate = jest.fn()
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockUsedNavigate,
}))

describe('Topic component', () => {
  test('component render', () => {
    const topic = render(<ForumTopicPage />)

    expect(topic).toMatchSnapshot()
  })
})
