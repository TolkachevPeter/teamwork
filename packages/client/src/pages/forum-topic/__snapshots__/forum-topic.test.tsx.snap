// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topic component component render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="forum-topic__container flex flex-column flex-ai-center"
      >
        <div
          class="topic__container flex flex-column"
        >
          <h3
            class="topic__header"
          >
            Заголовок топика1
          </h3>
          <p
            class="topic__text text-base-font-regular"
          >
            Полный текст топика Полный текст топика Полный текст топика Полный текст топика Полный текст топика Полный текст топика Полный текст топика Полный текст топика Полный текст топика Полный текст топика Полный текст топика Полный текст топика Полный текст топика
          </p>
        </div>
        <h5
          class="comments-header text-base-font-bold"
        >
          Комментарии:
        </h5>
        <div
          class="comment__container flex flex-column"
        >
          <div
            class="comment__header flex flex-jc-sb"
          >
            <h3
              class="comment__username text-xl-font-bold"
            >
              Иван
            </h3>
            <span
              class="comment__date"
            >
              06.08.2023
            </span>
          </div>
          <p
            class="comment__text text-base-font-regular"
          >
            Игра топ
          </p>
        </div>
        <div
          class="comment__container flex flex-column"
        >
          <div
            class="comment__header flex flex-jc-sb"
          >
            <h3
              class="comment__username text-xl-font-bold"
            >
              Коля
            </h3>
            <span
              class="comment__date"
            >
              06.08.2023
            </span>
          </div>
          <p
            class="comment__text text-base-font-regular"
          >
            Команда красавцы
          </p>
        </div>
        <form
          class="new-comment-form form__container flex flex-column"
        >
          <h5
            class="text-xl-font-bold"
          >
            Оставить комментарий
          </h5>
          <textarea
            class="comments__textarea text-base-font-regular"
          />
          <button
            class="button add-comment-button"
            name="addCommentBtn"
          >
            Отправить
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="forum-topic__container flex flex-column flex-ai-center"
    >
      <div
        class="topic__container flex flex-column"
      >
        <h3
          class="topic__header"
        >
          Заголовок топика1
        </h3>
        <p
          class="topic__text text-base-font-regular"
        >
          Полный текст топика Полный текст топика Полный текст топика Полный текст топика Полный текст топика Полный текст топика Полный текст топика Полный текст топика Полный текст топика Полный текст топика Полный текст топика Полный текст топика Полный текст топика
        </p>
      </div>
      <h5
        class="comments-header text-base-font-bold"
      >
        Комментарии:
      </h5>
      <div
        class="comment__container flex flex-column"
      >
        <div
          class="comment__header flex flex-jc-sb"
        >
          <h3
            class="comment__username text-xl-font-bold"
          >
            Иван
          </h3>
          <span
            class="comment__date"
          >
            06.08.2023
          </span>
        </div>
        <p
          class="comment__text text-base-font-regular"
        >
          Игра топ
        </p>
      </div>
      <div
        class="comment__container flex flex-column"
      >
        <div
          class="comment__header flex flex-jc-sb"
        >
          <h3
            class="comment__username text-xl-font-bold"
          >
            Коля
          </h3>
          <span
            class="comment__date"
          >
            06.08.2023
          </span>
        </div>
        <p
          class="comment__text text-base-font-regular"
        >
          Команда красавцы
        </p>
      </div>
      <form
        class="new-comment-form form__container flex flex-column"
      >
        <h5
          class="text-xl-font-bold"
        >
          Оставить комментарий
        </h5>
        <textarea
          class="comments__textarea text-base-font-regular"
        />
        <button
          class="button add-comment-button"
          name="addCommentBtn"
        >
          Отправить
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
